site/app/views/NavigationView.php

```diff
@@ -139,13 +139,55 @@ public function showGradeables($sections_to_list) {
         <tr class="colspan nav-title-row" id="{$lower_title}"><td colspan="4">{$title_to_category_title[$title]}</td></tr>
         <tbody id="{$lower_title}_tbody">
 HTML;
+            $title_save = $title;
+            $btn_title_save = $title_to_button_type_submission[$title];
             foreach ($gradeable_list as $gradeable => $g_data) {
-
+                $title = $title_save;
+                $title_to_button_type_submission[$title_save] = $btn_title_save;
                 // student users should only see electronic gradeables -- NOTE: for now, we might change this design later
                 if ($g_data->getType() != GradeableType::ELECTRONIC_FILE && !$this->core->getUser()->accessGrading()) {
                   continue;
                 }
 
+
+                if ($g_data->getActiveVersion() < 1){
+                    if ($title == "GRADED" || $title == "ITEMS BEING GRADED"){
+                        $title = "CLOSED";
+                    }
+                }
+
+                if ($g_data->beenAutograded() && $g_data->beenTAgraded()){
+                    
+                    $user_id = $this->core->getUser()->getId();
+                    $g_id = $g_data->getId();
+                    
+                    
+                    $params = array($user_id, $g_id);
+
+                    //A string representation of the sql query
+                    $query = "SELECT 
+                    user_viewed_date
+                    FROM
+                        gradeable_data 
+                    WHERE
+                        gd_user_id = ?
+                    AND 
+                        g_id = ?
+                    ;";
+
+
```

Comment:
Please remove these multiple empty lines between code blocks. We really only need one empty line to separate out things (and there's one case where you have 4+ empty lines between blocks which should definitely be reduced).